Automattic/jetpack
# Sharing: ask search engines to not index pages with query string.

Fixes #21836

#### Changes proposed in this Pull Request:

Jetpack's unofficial (icon, icon+text, text) sharing buttons currently work like so:
- When you append a query string with the name of a sharing service to a post URL, we redirect to the social service in question, with information the service can use (e.g. the post title as a tweet message).
- That query string may look like `?share=twitter`
- A page with such query string is not directly accessible since you are immediately redirected to the social network.
- WP Core implements `canonical` rel links in the head of each post page, so if you were looking at such a page's source code before the redirect happens, you would see that the canonical link for that page is in fact without the query string.
- Our HTML markup for the sharing buttons includes a `nofollow` rel attribute in the links, so as to indicate to search engines that the link should not be followed.

Yet that does not seem to be enough, as Google Search Console still seems to complain about those links in some scenarios, as mentioned in the issue above.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Go to Settings > Reading and ensure your site is publicly accessible and indexable.
* Activate the Sharing module on your site.
* In Settings > Sharing, ensure the Twitter button is added.
* Load a published post or a page with sharing buttons.
* Click on the Twitter button
* Ensure that you are properly redirected, that you can share the post.
* Back to the post, copy the URL of the button instead of clicking on it.
* In your terminal, run `curl -I the-url-you-copied` and ensure that the response includes the following: `X-Robots-Tag: noindex, nofollow`
* Try with a bunch of other services, not just Twitter.
